junitparser/junitparser.py

```diff
@@ -344,35 +344,37 @@ def is_passed(self):
         """Whether this testcase was a success (i.e. if it isn't skipped, failed, or errored)."""
         return not self.result
 
+    @property
+    def is_failure(self):
+        """Whether this testcase failed."""
+        return any(isinstance(r, Failure) for r in self.result)
+
+    @property
+    def is_error(self):
+        """Whether this testcase errored."""
+        return any(isinstance(r, Error) for r in self.result)
+
     @property
     def is_skipped(self):
         """Whether this testcase was skipped."""
-        for r in self.result:
-            if isinstance(r, Skipped):
-                return True
-        return False
+        return any(isinstance(r, Skipped) for r in self.result)
 
     @property
     def result(self):
         """A list of :class:`Failure`, :class:`Skipped`, or :class:`Error` objects."""
-        results = []
-        for entry in self:
-            if isinstance(entry, tuple(POSSIBLE_RESULTS)):
-                results.append(entry)
-
-        return results
+        return [r for r in self if isinstance(r, tuple(POSSIBLE_RESULTS))]
```

Comment:
This might be a breaking change. This method used to return a `list`, now, I think, it returns a generator. User code like `case.result[0]` will fail. We should either `return list([...])`, or release this as a breaking change (together with #142).

Reply:
this is a list-comprehention which constructs a list in-place, a generator expression would be surrounded by `( .. )`

this shows the differences:

```python
list_comprehentions = [x for x in range(10)]
print(f"{type(list_comprehentions) = }")
try:
    print(list_comprehentions[1])
except Exception:
    print("can't index 'list_comprehentions'")

dict_comprehentions = {x: x for x in range(10)}
print(f"{type(dict_comprehentions) = }")
try:
    print(dict_comprehentions[1])
except Exception:
    print("can't index 'dict_comprehentions'")

generator_expression = (x for x in range(10))
print(f"{type(generator_expression) = }")
try:
    print(generator_expression[1])
except Exception:
    print("can't index 'generator_expression'")
```

which results in:

```
type(list_comprehentions) = <class 'list'>
1
type(dict_comprehentions) = <class 'dict'>
1
type(generator_expression) = <class 'generator'>
can't index 'generator_expression'
```

Reply:
excellent, then this is safe to change